x-pack/legacy/plugins/uptime/server/lib/adapters/pings/adapter_types.ts

```diff
@@ -6,35 +6,83 @@
+import { UMElasticsearchQueryFn } from '../framework';
 
+export interface GetAllParams {
+  /** @member dateRangeStart timestamp bounds */
+  dateRangeStart: string;
+
+  /** @member dateRangeEnd timestamp bounds */
+  dateRangeEnd: string;
+
+  /** @member monitorId optional limit by monitorId */
+  monitorId?: string | null;
+
+  /** @member status optional limit by check statuses */
+  status?: string | null;
+
+  /** @member sort optional sort by timestamp */
+  sort?: string | null;
+
+  /** @member size optional limit query size */
+  size?: number | null;
+
+  /** @member location optional location value for use in filtering*/
+  location?: string | null;
+}
+
+export interface GetLatestMonitorDocsParams {
+  /** @member dateRangeStart timestamp bounds */
+  dateRangeStart: string;
+
+  /** @member dateRangeEnd timestamp bounds */
+  dateRangeEnd: string;
+
+  /** @member monitorId optional limit to monitorId */
+  monitorId?: string | null;
+
+  /** @member location optional location value for use in filtering*/
+  location?: string | null;
+}
+
+export interface GetPingHistogramParams {
+  /** @member dateRangeStart timestamp bounds */
+  dateRangeStart: string;
+  /** @member dateRangeEnd timestamp bounds */
+  dateRangeEnd: string;
+  /** @member filters user-defined filters */
+  filters?: string | null;
+  /** @member monitorId optional limit to monitorId */
+  monitorId?: string | null;
+  /** @member statusFilter special filter targeting the latest status of each monitor */
+  statusFilter?: string | null;
+}
+
+/**
+ * Count the number of documents in heartbeat indices
+ */
 export interface UMPingsAdapter {
-  getAll(
-    request: any,
-    dateRangeStart: string,
-    dateRangeEnd: string,
-    monitorId?: string | null,
-    status?: string | null,
-    sort?: string | null,
-    size?: number | null,
-    location?: string | null
-  ): Promise<PingResults>;
-
-  getLatestMonitorDocs(
-    request: any,
-    dateRangeStart: string,
-    dateRangeEnd: string,
-    monitorId?: string | null,
-    location?: string | null
-  ): Promise<Ping[]>;
-
-  getPingHistogram(
-    request: any,
-    dateRangeStart: string,
-    dateRangeEnd: string,
-    filters?: string | null,
-    monitorId?: string | null,
-    statusFilter?: string | null
-  ): Promise<HistogramResult>;
-
-  getDocCount(request: any): Promise<DocCount>;
+  getAll: UMElasticsearchQueryFn<GetAllParams, PingResults>;
+
+  getLatestMonitorDocs: UMElasticsearchQueryFn<GetLatestMonitorDocsParams, Ping[]>;
+
+  getPingHistogram: UMElasticsearchQueryFn<GetPingHistogramParams, HistogramResult>;
+
+  /**
+   * Gets data used for a composite histogram for the currently-running monitors.
+   */
+  getDocCount: UMElasticsearchQueryFn<{}, DocCount>;
+}
+
+export interface HistogramQueryResult {
+  key: number;
+  doc_count: number;
+  bucket_total: {
+    value: number;
+  };
+  down: {
+    bucket_count: {
+      value: number;
+    };
+  };
 }
```

Comment:
Another unrelated to PR , suggestion is to rename adapter_types file to just types.

Reply:
👍 this naming convention predates Uptime and was a result of mirroring other new plugins. I think we've done away with many of the conventions, so I am fine with renaming these files to make them less verbose.

Reply:
I created https://github.com/elastic/uptime/issues/126